### PR TITLE
Add Debian 11: Bulleyes to support systems check

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds `Debian GNU/Linux 11 (bullseye)` to the list of supported operating systems.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/19031
- Link to cli pull request:
- Other: https://github.com/home-assistant/supervisor/pull/3066

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by ludeeus — APPROVED** [review]
The [ADR](https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md)  and [documentation](https://www.home-assistant.io/more-info/unsupported/os) needs an adjustment as well

File: supervisor/resolution/evaluations/operating_system.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from ..const import UnsupportedReason
 from .base import EvaluateBase
 
-SUPPORTED_OS = ["Debian GNU/Linux 10 (buster)"]
+SUPPORTED_OS = ["Debian GNU/Linux 10 (buster)", "Debian GNU/Linux 11 (bullseye)"]
```
Comment (pvizeli) [inline supervisor/resolution/evaluations/operating_system.py:9]: We support only one version. I don't want run 2 debian version or support sub deps in any kind of version

**Review by pvizeli — CHANGES_REQUESTED** [review]

File: supervisor/resolution/evaluations/operating_system.py (hunk 1 above)
Comment (frenck) [reply supervisor/resolution/evaluations/operating_system.py:9]: According to the ADR we have a deprecation period of 4 months.

> When a new major version of Debian is released, the previous major version is dropped, with a deprecation time of 4 months

<https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md>

Which means a 4 month overlap period.

**Review by pvizeli — APPROVED** [review]
Right, I had that wrong in the mind

**Comment by frenck** [comment]
ADR PR: https://github.com/home-assistant/supervisor/pull/3066
Docs PR: https://github.com/home-assistant/home-assistant.io/pull/19031

**Pull request merged**